src/Utils/Utils.php

```diff
@@ -149,15 +150,23 @@ public static function getCleanShopwareVersion(): string
      */
     protected static function getShopwareVersion(): string
     {
-        $packageVersions = Versions::VERSIONS;
-        $coreIsInstalled = isset($packageVersions['shopware/core']);
-
-        $shopwareVersion = $coreIsInstalled ?
-            Versions::getVersion('shopware/core') :
-            Versions::getVersion('shopware/platform');
-
-        if (!trim($shopwareVersion, 'v@')) {
-            return $coreIsInstalled ? $packageVersions['shopware/core'] : $packageVersions['shopware/platform'];
+        // Composer 2 runtime API uses the `InstalledVersions::class` in favor of the
+        // deprecated/removed `Versions::class`
+        if (class_exists(InstalledVersions::class)) {
+            $platformIsInstalled = InstalledVersions::isInstalled('shopware/platform');
+            if ($platformIsInstalled) {
+                $shopwareVersion = InstalledVersions::getPrettyVersion('shopware/platform');
+            } else {
+                $shopwareVersion = InstalledVersions::getPrettyVersion('shopware/core');
+            }
```

Comment:
You could also return the versions here directly, which would result in the `else` block in line 162 to no longer be necessary.